crates/sui-benchmark/src/workloads/batch_payment.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_trait::async_trait;
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use sui_types::{
+    base_types::{ObjectRef, SuiAddress},
+    crypto::get_key_pair,
+    messages::VerifiedTransaction,
+};
+
+use crate::in_memory_wallet::InMemoryWallet;
+use crate::system_state_observer::SystemStateObserver;
+use crate::workloads::payload::Payload;
+use crate::workloads::{Gas, GasCoinConfig, WorkloadInitGas, WorkloadPayloadGas};
+use crate::{ExecutionEffects, ValidatorProxy};
+use sui_core::test_utils::make_pay_sui_transaction;
+
+use super::workload::{Workload, WorkloadType, MAX_GAS_FOR_TESTING};
+
+#[derive(Debug)]
+pub struct BatchPaymentTestPayload {
+    state: InMemoryWallet,
+    system_state_observer: Arc<SystemStateObserver>,
+}
+
+impl Payload for BatchPaymentTestPayload {
+    fn make_new_payload(self: Box<Self>, effects: &ExecutionEffects) -> Box<dyn Payload> {
+        let mut state = self.state;
+        state.update(effects);
+        Box::new(BatchPaymentTestPayload {
+            state,
+            system_state_observer: self.system_state_observer,
+        })
+    }
+
+    fn make_transaction(&self) -> VerifiedTransaction {
+        let addrs = self.state.addresses().cloned().collect::<Vec<SuiAddress>>();
+        let num_recipients = addrs.len();
+        // TODO: round-robin once we have mutability
+        let sender = addrs[0];
+        // create a sender -> all transfer
+        let objs = self
+            .state
+            .owned_objects(&sender)
+            .unwrap()
+            .cloned()
+            .collect::<Vec<ObjectRef>>();
+        // we're only using gas objects in this benchmark, so safe to assume everything owned by an address is a gas object
+        let gas_obj = objs[0];
+        // TODO: be smarter about this?
+        let amount = 1;
+        make_pay_sui_transaction(
+            gas_obj,
+            objs,
+            addrs,
+            vec![amount; num_recipients],
+            sender,
+            &self.state.keypair(&sender).unwrap(),
+            Some(*self.system_state_observer.reference_gas_price.borrow()),
+        )
+    }
+
+    fn get_workload_type(&self) -> WorkloadType {
+        WorkloadType::BatchPayment
+    }
+}
+
+#[derive(Debug)]
+pub struct BatchPaymentWorkload {
+    batch_size: u64,
+}
+
+impl BatchPaymentWorkload {
+    pub fn new() -> Self {
+        // TODO: make this configurable via CLI
+        BatchPaymentWorkload { batch_size: 100 }
```

Comment:
@tnowacki seeking your guidance on how big I can make this before I hit limits.

Reply:
Not sure exactly how this manifests itself without digging in some more. But at a high level, the max number of commands is currently at 1024 and the max number of args is 512